### PR TITLE
feat: add version command

### PR DESCRIPTION
**Pull request by jwillker**
The version command was missing, as mentioned in this [issue](https://github.com/google/osv-scanner/issues/42).

The goreleaser already setting build information here: https://github.com/google/osv-scanner/blob/e5f1dc67baff39c840963a5966219f81e2c7fe76/.goreleaser.yml#L14-L16, to take advantage of this I implemented this feature based on this information.

The "urfave/cli" already has a clean way to implement this feature, as mentioned here: https://cli.urfave.org/v2/examples/version-flag/

### CODE REVIEW

**Comment by wolf99** [comment]
Relates to #18

**Comment by oliverchang** [comment]
Thank you so much for contributing this!

File: cmd/osv-scanner/main.go [hunk 1]
```diff
@@ -12,11 +12,23 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+var (
+	version = ""
```
Comment (another-rex) [inline cmd/osv-scanner/main.go:16]: Can you add "`dev`" to version here as a default, and `n/a` for commit and date?

File: cmd/osv-scanner/main.go [hunk 2]
```diff
@@ -12,11 +12,23 @@ import (
...
+	commit  = ""
+	date    = ""
+)
+
 func run(args []string, stdout, stderr io.Writer) int {
 	var r *output.Reporter
 
+	cli.VersionPrinter = func(ctx *cli.Context) {
+		r = output.NewReporter(stdout, stderr, false)
+		r.PrintText(fmt.Sprintf("osv-scanner version %s\ncommit: %s\nbuilt at: %s\n", ctx.App.Version, commit, date))
```
Comment (another-rex) [inline cmd/osv-scanner/main.go:26]: Add a `:` after `osv-scanner version`

**Review by another-rex — COMMENTED** [review]
Thanks for the PR!, Just two minor comments.

File: cmd/osv-scanner/main.go (hunk 1 above)
Comment (jwillker) [reply cmd/osv-scanner/main.go]: Done!

File: cmd/osv-scanner/main.go (hunk 2 above)
Comment (jwillker) [reply cmd/osv-scanner/main.go]: Done!

**Review by another-rex — APPROVED** [review]

**Pull request merged**